### PR TITLE
Analytics fixes + alt description for google images

### PR DESCRIPTION
**Pull request by marjev**
@martynjames latest changes regarding comments @cptvitamin and @mulby wrote, please review

### CODE REVIEW

**Comment by mulby** [comment]
:+1: from an analytics code standpoint

Can you post samples of each of these events from the tracking.log file in your devstack? It will help @stroilova review them from a researcher perspective and help @lamagnifica document them.

File: google_drive/public/js/google_calendar.js [hunk 1]
```diff
@@ -1,12 +1,15 @@
 /* Javascript for GoogleDocumentBlock. */
 function GoogleCalendarBlock(runtime, element) {
 
+    var ajaxflag = $('.ajaxflag', element);
```
Comment (martynjames) [inline google_drive/public/js/google_calendar.js]: I have no idea what "```ajaxflag```" is! It appears to exist solely for the purpose of altering content upon completion of publishing the event; I assume this is for testing purposes, so that one can say something along the lines of:
"I've published the event that indicates that the load has completed".

There's a couple of things of note herein:
i) The name is meaningless - I'd choose something more meaningful, such as ```load_event_complete```
ii) The variable is only used once, seems to me that it would be much easier to read the code if the selector was simply used in the one place in which it is used.

File: google_drive/google_calendar.py [hunk 2]
```diff
@@ -42,7 +42,7 @@ class GoogleCalendarBlock(XBlock, PublishEventMixin):
 
     views = [(0, 'Week'), (1, 'Month'), (2, 'Agenda')]
 
-    def student_view(self, context={}):
+    def student_view(self, context):
```
Comment (martynjames) [inline google_drive/google_calendar.py]: Note that ```context``` is now unused when passed into student_view

File: google_drive/public/js/google_docs.js [hunk 3]
```diff
@@ -26,12 +29,20 @@ function GoogleDocumentBlock(runtime, element) {
...
+            }),
+            success: function(){
+                ajaxflag.val("1");
+            }
         });
     }
-    iframe.load(function(e){SignalDocumentLoaded(e, 'iframe');});
-    image.load(function(e){SignalDocumentLoaded(e, 'img');});
+
+    $('iframe', element).load(function(e){
```
Comment (martynjames) [inline google_drive/public/js/google_docs.js]: I don't quite understand this change - we've already made the selection above, yet we are calling the selector again - what's the purpose of this change?

File: google_drive/google_calendar.py (hunk 2 above)
Comment (marjev) [inline google_drive/google_calendar.py]: Workbench had a problem with this part of the code, as if it didn't set context to an empty dictionary, thus throwing an error that a None object doesn't have a method .update().

File: google_drive/public/js/google_docs.js (hunk 3 above)
Comment (marjev) [inline google_drive/public/js/google_docs.js]: I made this change while I was writing tests, but obviously forgot to change it back. I will revert this change.

**Comment by marjev** [comment]
Here are the samples from tracker.log, as requested @mulby @stroilova @lamagnifica

**Calendar:** 
```
{"username": "staff", "host": "precise64", "event_source": "server", "event_type": "/courses/GoogleDrive/GD2015/2015_GGL/xblock/i4x:;_;_GoogleDrive;_GD2015;_google-calendar;_fddbefc49c144577880e8edcb35a7f26/handler/publish_event", "context": {"course_user_tags": {}, "user_id": 4, "org_id": "GoogleDrive", "course_id": "GoogleDrive/GD2015/2015_GGL", "path": "/courses/GoogleDrive/GD2015/2015_GGL/xblock/i4x:;_;_GoogleDrive;_GD2015;_google-calendar;_fddbefc49c144577880e8edcb35a7f26/handler/publish_event"}, "time": "2015-01-07T09:38:49.478994+00:00", "ip": "10.0.2.2", "event": "{\"POST\": {\"{\\\"url\\\":\\\"https://www.google.com/calendar/embed?mode\": [\"Agenda\"], \"showCalendars\": [\"0\\\",\\\"displayedin\\\":\\\"iframe\\\",\\\"event_type\\\":\\\"edx.googlecomponent.calendar.displayed\\\"}\"], \"src\": [\"edx.org_lom804qe3ttspplj1bgeu1l3ak@group.calendar.google.com\"]}, \"GET\": {}}", "agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/39.0.2171.71 Safari/537.36", "page": null}
{"username": "staff", "host": "precise64", "event_source": "server", "event_type": "edx.googlecomponent.calendar.displayed", "context": {"course_user_tags": {}, "user_id": 4, "org_id": "GoogleDrive", "module": {"display_name": "Google Calendar"}, "course_id": "GoogleDrive/GD2015/2015_GGL", "path": "/courses/GoogleDrive/GD2015/2015_GGL/xblock/i4x:;_;_GoogleDrive;_GD2015;_google-calendar;_fddbefc49c144577880e8edcb35a7f26/handler/publish_event"}, "time": "2015-01-07T09:38:49.536840+00:00", "ip": "10.0.2.2", "event": {"url": "https://www.google.com/calendar/embed?mode=Agenda&src=edx.org_lom804qe3ttspplj1bgeu1l3ak@group.calendar.google.com&showCalendars=0", "displayedin": "iframe"}, "agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/39.0.2171.71 Safari/537.36", "page": "x_module"}
```

**Document:**
```
{"username": "staff", "host": "precise64", "event_source": "server", "event_type": "/courses/GoogleDrive/GD2015/2015_GGL/xblock/i4x:;_;_GoogleDrive;_GD2015;_google-document;_1a1a7b8cd3c04c2e8c906270b5cadbc7/handler/publish_event", "context": {"course_user_tags": {}, "user_id": 4, "org_id": "GoogleDrive", "course_id": "GoogleDrive/GD2015/2015_GGL", "path": "/courses/GoogleDrive/GD2015/2015_GGL/xblock/i4x:;_;_GoogleDrive;_GD2015;_google-document;_1a1a7b8cd3c04c2e8c906270b5cadbc7/handler/publish_event"}, "time": "2015-01-07T09:38:49.379322+00:00", "ip": "10.0.2.2", "event": "{\"POST\": {\"delayms\": [\"10000\\\",\\\"displayedin\\\":\\\"iframe\\\",\\\"event_type\\\":\\\"edx.googlecomponent.document.displayed\\\"}\"], \"loop\": [\"true\"], \"{\\\"url\\\":\\\"https://docs.google.com/presentation/d/1x2ZuzqHsMoh1epK8VsGAlanSo7r9z55ualwQlj-ofBQ/embed?start\": [\"true\"]}, \"GET\": {}}", "agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/39.0.2171.71 Safari/537.36", "page": null}
{"username": "staff", "host": "precise64", "event_source": "server", "event_type": "edx.googlecomponent.document.displayed", "context": {"course_user_tags": {}, "user_id": 4, "org_id": "GoogleDrive", "module": {"display_name": "Google Document"}, "course_id": "GoogleDrive/GD2015/2015_GGL", "path": "/courses/GoogleDrive/GD2015/2015_GGL/xblock/i4x:;_;_GoogleDrive;_GD2015;_google-document;_1a1a7b8cd3c04c2e8c906270b5cadbc7/handler/publish_event"}, "time": "2015-01-07T09:38:49.428596+00:00", "ip": "10.0.2.2", "event": {"url": "https://docs.google.com/presentation/d/1x2ZuzqHsMoh1epK8VsGAlanSo7r9z55ualwQlj-ofBQ/embed?start=true&loop=true&delayms=10000", "displayedin": "iframe"}, "agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/39.0.2171.71 Safari/537.36", "page": "x_module"}
```

**Image:**
```
{"username": "staff", "host": "precise64", "event_source": "server", "event_type": "/courses/GoogleDrive/GD2015/2015_GGL/xblock/i4x:;_;_GoogleDrive;_GD2015;_google-document;_b4aeff80bb7641caae60c10713a87617/handler/publish_event", "context": {"course_user_tags": {}, "user_id": 4, "org_id": "GoogleDrive", "course_id": "GoogleDrive/GD2015/2015_GGL", "path": "/courses/GoogleDrive/GD2015/2015_GGL/xblock/i4x:;_;_GoogleDrive;_GD2015;_google-document;_b4aeff80bb7641caae60c10713a87617/handler/publish_event"}, "time": "2015-01-07T09:54:15.440976+00:00", "ip": "10.0.2.2", "event": "{\"POST\": {\"h\": [\"657\\\",\\\"displayedin\\\":\\\"img\\\",\\\"event_type\\\":\\\"edx.googlecomponent.document.displayed\\\"}\"], \"{\\\"url\\\":\\\"https://docs.google.com/drawings/d/1LHGzCTLRb--CDvFFjoYp62TiIN5KgsE7QOy9Sift_eg/pub?w\": [\"882\"]}, \"GET\": {}}", "agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/39.0.2171.71 Safari/537.36", "page": null}
{"username": "staff", "host": "precise64", "event_source": "server", "event_type": "edx.googlecomponent.document.displayed", "context": {"course_user_tags": {}, "user_id": 4, "org_id": "GoogleDrive", "module": {"display_name": "Google Document"}, "course_id": "GoogleDrive/GD2015/2015_GGL", "path": "/courses/GoogleDrive/GD2015/2015_GGL/xblock/i4x:;_;_GoogleDrive;_GD2015;_google-document;_b4aeff80bb7641caae60c10713a87617/handler/publish_event"}, "time": "2015-01-07T09:54:15.497526+00:00", "ip": "10.0.2.2", "event": {"url": "https://docs.google.com/drawings/d/1LHGzCTLRb--CDvFFjoYp62TiIN5KgsE7QOy9Sift_eg/pub?w=882&h=657", "displayedin": "img"}, "agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/39.0.2171.71 Safari/537.36", "page": "x_module"}
```

**Comment by mulby** [comment]
@marjev so the sample events you posted have actually highlighted an issue. Because you implemented a custom event handling endpoint, we will actually see 2 different events in the tracking log. This is because the tracking middleware logs *every* request made to the LMS to the tracking log except for a small number of endpoints which are explicitly ignored. We should probably add your endpoint to that list.

The events we want to see should have an event_type that looks like "edx.googlecomponent.*".

https://github.com/edx/edx-platform/blob/master/common/djangoapps/track/middleware.py#L79

https://github.com/edx/edx-platform/blob/master/lms/envs/common.py#L514

**Comment by marjev** [comment]
@mulby doesn't the second event (second line) in each example have the event_type like that, for example, the calendar has "edx.googlecomponent.calendar.displayed" for the event_type? I copy-pasted the first event just because it states that the unit containing my xblock was published, but you can ignore the first event, if it's confusing.

If you wish, I can edit my comment and remove the first event.

**Comment by mulby** [comment]
@marjev ah, sorry I missed that, I was reading it on my mobile phone and only read the first event.

Please remove the "/courses/..." event types from the sample events. It will be easier for Olga and Alison to review.

I'm OK with holding off on addressing the duplicate event issue since the analytics team is planning to take a look at a holistic solution to JS event tracking from XBlocks in the near future. We can always come back in and migrate this code over to that new system if/when it is complete.

**Comment by marjev** [comment]
@mulby Changed my comment. If anything changes from analytics standpoint, let me know. Thanks!

**Comment by stroilova** [comment]
Hi @martynjames I see in the comments above that 'displayedin' was changed to 'displayed_in', but the sample events linked have 'displayedin' as a field. 
What is the correct name for this field? Thank you.

Also, @martynjames  , @mulby  / @shnayder  , these events seem to not have a 'name' field filled out. 
'event_type' is a deprecated field that is maintained for backwards-compatibility. We suggest developers fill out both 'name' and 'event_type' to the same value for newly emitted events.

My apologies for catching this only now!

I know this PR has been merged to master, but I did want to double-check before releasing the documentation as well. 

If it is not too late, could updated sample events be posted here as a comment?

Ref: https://github.com/edx/edx-documentation/pull/160

FYI: @lamagnifica ,

**Comment by martynjames** [comment]
I seem to have seen a comment pointing out the change to displayed_in for the documentation review. Yes, I believe that we made that change.

I will log a new ticket for adding the ‘name’. It seems better to post sample events in the PR for that update, so I’ll plan to have those put in there.


Thanks,

Marty

**Comment by stroilova** [comment]
Thank you @martynjames , much appreciated!! 

I'm sharing a google doc with you with events to review for validation on Stage.
